koku/cost_models/view.py

```
@@ -174,12 +175,11 @@ def destroy(self, request, *args, **kwargs):
         uuidParam = kwargs.get("uuid")
         try:
             manager = CostModelManager(cost_model_uuid=uuidParam)
-        except CostModel.DoesNotExist:
+            manager.update_provider_uuids([])
+        except (CostModel.DoesNotExist, CostModelException):
             LOG.info("CostModel does not exist.")
         except ValidationError as err:
             raise CostModelQueryException(err)
```

Ruff (0.16.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)